app/beacon-omop-worker/beacon_omop_worker/query_solvers.py

```diff
@@ -1,22 +1,110 @@
 from typing import List
 from beacon_omop_worker.beacon_dto.filtering_term import FilteringTerm
+from beacon_omop_worker.beacon_dto.response_summary import ResponseSummary
 from beacon_omop_worker.db_manager import SyncDBManager
 from beacon_omop_worker.entities import (
     ConditionOccurrence,
     Person,
-    DrugExposure,
-    Measurement,
-    Observation,
-    ProcedureOccurrence,
     Concept,
     Vocabulary,
 )
 import beacon_omop_worker.config as config
 import logging
 import pandas as pd
-from sqlalchemy import (
-    select,
-)
+from sqlalchemy import select, and_
+
+
+class IndividualQuerySolver:
+
+    def __init__(self, db_manager: SyncDBManager) -> None:
+        self.db_manager = db_manager
+
+    def _get_concept_id(self, vocab: str, concept_code: str) -> str:
+        """
+        Get concept id from vocabulary and concept code
+        Args:
+            vocab (str): vocabulary name
+            concept_code (str): concept code
+
+        Returns:
+
+        """
+        query = select(Concept.concept_id).where(
+            and_(
+                Concept.vocabulary_id == vocab,
+                Concept.concept_code == concept_code,
+            )
+        )
+        code = pd.read_sql(query, con=self.db_manager.engine.connect())
```

Comment:
I'm not sure you need to be calling `connect` all the time on `self.db_manager.engine`. If you look at the `query_solvers.py` in `rquest-omop-worker`, it just gives the engine without calling `connect`.

Reply:
latest version of sqlalchemy (>2.0) removed engine.execute so that doesn't work anymore, it has to call connect()